azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -435,31 +399,150 @@ def test_send_method_response_calls_pipeline(self, client, pipeline):
         "Waits for the completion of the 'send_method_response' pipeline operation before returning"
     )
     def test_waits_for_pipeline_op_completion(self, mocker, client_manual_cb, pipeline_manual_cb):
-        client = client_manual_cb
-        pipeline = pipeline_manual_cb
-        event_init_mock = mocker.patch.object(threading, "Event")
-        event_mock = event_init_mock.return_value
+        self.add_event_completion_checks(
+            mocker=mocker, pipeline_function=pipeline_manual_cb.send_method_response
+        )
+        response = MethodResponse(request_id="1", status=200, payload={"key": "value"})
+        client_manual_cb.send_method_response(response)
 
-        def check_callback_completes_event():
-            # Assert exactly one Event was instantiated so we know the following asserts
-            # are related to the code under test ONLY
-            assert event_init_mock.call_count == 1
 
-            # Assert waiting for Event to complete
-            assert event_mock.wait.call_count == 1
-            assert event_mock.set.call_count == 0
+class SharedClientGetTwinTests(WaitsForEventCompletion):
+    @pytest.mark.it("Begins a 'get_twin' pipeline operation")
+    def test_get_twin_calls_pipeline(self, client, pipeline):
+        client.get_twin()
+        assert pipeline.get_twin.call_count == 1
 
-            # Manually trigger callback
-            cb = pipeline.send_method_response.call_args[1]["callback"]
-            cb()
+    @pytest.mark.it(
+        "Waits for the completion of the 'get_twin' pipeline operation before returning and returns the twin that the pipeline returned"
```

Comment:
>"Waits for the completion of the 'get_twin' pipeline operation before returning and returns the twin that the pipeline returned" [](start = 8, length = 128)

IMO these are two different tests/requirements. One that shows it waits on the operation, one that shows what it returns.